src/models/ctx/update_streams.rs

```diff
@@ -42,12 +46,38 @@ pub fn update_streams<E: Env + 'static>(
                 video_id: video_id.to_owned(),
                 meta_transport_url: meta_request.base.to_owned(),
                 stream_transport_url: stream_request.base.to_owned(),
+                state: last_stream_state,
                 mtime: E::now(),
             };
 
             streams.items.insert(key, streams_item);
             Effects::msg(Msg::Internal(Internal::StreamsChanged(false)))
         }
+        Msg::Internal(Internal::StreamStateChanged {
+            state,
+            stream_request: Some(stream_request),
+            meta_request: Some(meta_request),
+        }) => {
+            let meta_id = &meta_request.path.id;
+            let video_id = &stream_request.path.id;
+
+            let key = StreamsItemKey {
+                meta_id: meta_id.to_owned(),
+                video_id: video_id.to_owned(),
+            };
+            let steam_item = streams.items.get(&key).cloned();
+            match steam_item {
+                Some(item) => {
+                    let new_stream_item = StreamsItem {
+                        state: Some(state.to_owned()),
+                        ..item
+                    };
+                    streams.items.insert(key, new_stream_item);
+                    Effects::msg(Msg::Internal(Internal::StreamsChanged(false)))
```

Comment:
Not sure I got you meaning, this was done the same when inserting stream item above. We need to store the changed state into storage so we need to launch this effect.

Reply:
instead of `get` and later `insert` you can use `entry()` and handle the case where it's `Vacant` entry and directly mutate the empty value in the HashMap:

https://doc.rust-lang.org/std/collections/hash_map/enum.Entry.html

It's a more idiomatic way of doing what you've done here. Not related to the business logic itself